ts/metrics/metric_cache_abstract.py

```diff
@@ -0,0 +1,462 @@
+"""
+Abstract metric cache class using ABC.
+Implemented in the case it is decided that another file format is better in the future.
+
+Currently, abstract class has the methods for getting a metric and adding a metric to the cache.
+"""
+import abc
+import logging
+
+import ts.metrics.metric_cache_errors as merrors
+from ts.metrics.dimension import Dimension
+from ts.metrics.metric import Metric
+from ts.metrics.metric_type_enums import MetricTypes
+
+
+class MetricCacheAbstract(metaclass=abc.ABCMeta):
+    def __init__(self, request_ids, model_name, file):
+        """
+        Constructor for MetricsCaching class
+
+            MetricsCaching class will hold Metric objects in a cache that can be edited and read from.
+            The class also has the ability to parse a yaml file into Metrics objects
+
+        Parameters
+        ----------
+        file: str
+            Name of file to be parsed
+
+        """
+        self.cache = {}
+        self.store = []
+        self.request_ids = request_ids
+        self.model_name = model_name
+        self.file = file
+
+    @staticmethod
+    def _check_matching_dims(key, dims) -> bool:
+        """
+        Check to see if key value string pair already in list of dim strings
+        """
+        is_present = False
+        for dim in dims:
+            key_dim = dim.split(":")[0]
+            if key == key_dim:
+                is_present = True
+        return is_present
+
+    def _add_or_update(
+        self,
+        name: str,
+        value: int or float,
+        req_id,
+        unit: str,
+        metric_type: MetricTypes = None,
```

Comment:
when should `metric_type` validly be `None`?

Reply:
you're right there should not be a possibility that `metric_type `should be `None` in this method, my mistake